learning_paths/api/v1/views.py

```diff
@@ -393,3 +395,42 @@ def post(self, request, *args, **kwargs):
             },
             status=status.HTTP_201_CREATED,
         )
+
+
+class LearningPathCourseEnrollmentView(APIView):
+    """API View to enroll a user in a course that's part of a learning path."""
+
+    permission_classes = [IsAuthenticated, IsAdminOrSelf]
+
+    def _get_learning_path(self, learning_path_key_str: str) -> LearningPath:
+        """
+        Get the learning path and verify the user has access and is enrolled.
+
+        :raises: Http404 if the learning path is not found or the user does not have access.
+        """
+        return get_object_or_404(
+            LearningPath.objects.get_paths_visible_to_user(self.request.user).filter(
+                is_enrolled=True
+            ),
+            key=learning_path_key_str,
+        )
+
+    def post(self, request, learning_path_key_str: str, course_key_str: str):
```

Comment:
Are we not allowing unenroll option?

Reply:
@pkulkark, for courses? No - we definitely don't want this here, as it would make things hard to track in the analytics.

For LPs - we added a setting to toggle it in the API, but the client does not want to allow users to self-unenroll, so it is off by default, and we are not implementing it in the MFE now.